### DISCUSSION TITLE
commute time run() and runParrellelApproximation()

### DISCUSSION THREAD

**Discussion by Xiaohui9607** [networkit/networkit#1290] [Q&A]
to reproduce: 
```
b = nx.Graph()
b.add_nodes_from([0,1,2,3])
b.add_weighted_edges_from([(0,1,1.),(1,2,0.1),(2,3,1.)])
b_nk = nk.nxadapter.nx2nk(b, 'weight')
b_nk.indexEdges()
reff01 = nx.resistance_distance(b,0,1,'weight',False) 
reff02 = nx.resistance_distance(b,0,2,'weight',False)
reff03 = nx.resistance_distance(b,0,3,'weight',False)

ctd = nk.distance.CommuteTimeDistance(b_nk, tol=0.01)
ctd.run()
ct01 = ctd.distance(0,1)**2/(2*b_nk.totalEdgeWeight())
ct02 = ctd.distance(0,2)**2/(2*b_nk.totalEdgeWeight())
ct03 = ctd.distance(0,3)**2/(2*b_nk.totalEdgeWeight())


ctd = nk.distance.CommuteTimeDistance(b_nk, tol=0.01)
ctd.runParallelApproximation()
ct01a = ctd.distance(0,1)**2/(2*b_nk.totalEdgeWeight())
ct02a = ctd.distance(0,2)**2/(2*b_nk.totalEdgeWeight())
ct03a = ctd.distance(0,3)**2/(2*b_nk.totalEdgeWeight())

```
I have
reff01 = 1, reff02 = 11, reff03 = 12
ct01 = 1, ct02 = 11, ct03 = 12
ct01a = 1, ct02a = 102.70, ct03a = 103.755

**Comment by fabratu** [comment]
Hi,

thanks for bringing this up. We are currently investigating the issue. It seems the error in the approximation for weighted graphs is (at least for some cases) indeed much higher than for unweighted graphs.

Something to note: especially for very small graphs the approximation might not lead to meaningful results due to the underlying dimension reduction in the computation. Rule of thumb: `#nodes > log(#nodes) / (tol^2)`. So for smaller instances it is advised to use exact computation (or you increase the tolerance).

**Comment by Xiaohui9607** [comment]
Hi,

I am actually going to run the algorithm in a pretty large graphs (15000 nodes with millions of edges).

So there will have no problem if I ran it on large graph?

**Comment by fabratu** [comment]
Currently hard to tell, for smaller instances the error sure is too high:

```
foodweb-baydry
NetworKit.Graph(n=128, m=2137)
rel. error (weighted): 83.09529925520671
rel. error (unweighted): 0.020824405156682994
```

We had to delay additional tests for bigger instances so far. Looking forward to have more info in the upcoming days.
